application/features/authentication.feature

```diff
@@ -294,3 +294,19 @@ Feature: Authentication
     Then the response status code should be 200
     And The user's current password should be marked as pwned
     And The user's password is expired
+
+  Scenario Outline: Successfully authenticating even if the WebAuthn MFA API is unusable
+    Given "shep_clark" has a valid WebAuthn MFA method
+      And I provide the following valid data:
+        | property  | value        |
+        | username  | shep_clark   |
+        | password  | govols!!!    |
+      And we have the <rightOrWrongPassword> for the WebAuthn MFA API
+    When I request "/authentication" be created
+    Then the response status code should be 200
+      And the response body should <containPublicKeyOrNot>
+
+    Examples:
+      | rightOrWrongPassword | containPublicKeyOrNot   |
+      | wrong password       | not contain "publicKey" |
```

Comment:
Is there a way to write this with a positive test? Verifying that the response body does **not** contain something is, to me, more fragile. Some unrelated failure could happen and we wouldn't know about it because many possible responses could not contain the token. I suppose the real test is the previous line ("response status code should be 200") so this may not be a big deal.

Reply:
Yeah, the 200 response code is the main test. The "doesn't contain publicKey" is sort of to verify that the WebAuthn MFA API call did in fact fail. I could revise it to specifically look for the WebAuthn MFA API entry, and verify that it doesn't have the publicKey, but this seemed sufficient to be effective.